lib/expression_parser.rb

```diff
@@ -28,10 +30,15 @@ def write_vm_code(vm_writer, symbol_table)
   end
 end
 
-Boolean = Struct.new(:value) do
+KeywordConstant = Struct.new(:value) do
   def write_vm_code(vm_writer, symbol_table)
-    vm_writer.write_push(:CONST, 0)
-    vm_writer.write_arithmetic(:NOT) if value == :TRUE
+    case value
+    when *[:TRUE, :FALSE]
```

Comment:
The splat is used to treat an array as its individual values (e.g. for passing positional arguments to a method), so when you _already know_ the contents of the array it’s simpler to just list the values directly:

```suggestion
    when :TRUE, :FALSE
```